cmake/HPX_DetectCppDialect.cmake

```diff
@@ -116,6 +112,9 @@ macro(hpx_detect_cpp_dialect)
       hpx_info("C++ mode assumed: C++11")
     endif()
 
+    hpx_add_config_cond_define(_HAS_AUTO_PTR_ETC 1)
+    hpx_add_config_cond_define(_SILENCE_ALL_CXX17_DEPRECATION_WARNINGS)
```

Comment:
Shouldn't this go in the `HPX_WITH_CXX17` only?

Reply:
Yes, we can do that.